pkg/kubelet/kubelet.go

```diff
@@ -1732,6 +1748,30 @@ func (kl *Kubelet) syncPod(ctx context.Context, updateType kubetypes.SyncPodType
 		return nil
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		// While resize is in progress, periodically call PLEG to update pod cache
+		resizeInProgress := false
+		for _, c := range pod.Spec.Containers {
+			if cs, ok := podutil.GetContainerStatus(apiPodStatus.ContainerStatuses, c.Name); ok {
+				if cs.Resources == nil {
+					continue
+				}
+				if diff.ObjectDiff(c.Resources.Limits, cs.Resources.Limits) != "" ||
+					diff.ObjectDiff(cs.ResourcesAllocated, cs.Resources.Requests) != "" {
+					resizeInProgress = true
+					break
+				}
+			}
+		}
+		if resizeInProgress == true {
+			runningPod := kubecontainer.ConvertPodStatusToRunningPod(kl.getRuntime().Type(), podStatus)
+			if err := kl.pleg.UpdateCache(&runningPod, pod.UID); err != nil {
```

Comment:
I am trying to understand the motivation for this section.  Can you explain what metric updating the PLEG cache here improved rather than wait for the next relist interval?  Is some information lost that I missed by not alerting the PLEG and therefore this invocation was required?  I ask because @mrunalp and a few others are looking at runtime-native watch support, and I would expect the runtime to be able to inotify back the resource change that may have happened without this intersection point.

Reply:
At the time of writing this code, we did not have a way for runtime to notify resource changes. Consequently, PLEG could not identify an event that would result in invocation of GetPodStatus which queries for actual resources that runtime returns in ContainerStatus (with my CRI extension). So this change is a moderately optimal way to trigger GetPodStatus and look for convergence of Spec.Resources and Status.Resources. If we have a way for runtime to notify actuation of resource change, then this would be unnecessary I think. But AFAIK upon discussions with @Random-Liu that might be a fairly big change.
Does that address your question?

Reply:
@mrunalp @haircommander we still need to verify that the runtime will actually return actual usage (not just desired usage).